cmd/util/flags/configflags/job_selection.go

```diff
@@ -3,34 +3,72 @@ package configflags
 import "github.com/bacalhau-project/bacalhau/pkg/config/types"
 
 var JobSelectionFlags = []Definition{
+	//
+	// Requester Job Selection Policy
+	//
+	// TODO: https://github.com/bacalhau-project/bacalhau/issues/2929
+	// The flags are currently settable on the requester, however the requester does not support a JobSelectionPolicy
+	// setting the flags is a noop until the above issue is fixed.
 	{
-		FlagName:     "job-selection-data-locality",
+		FlagName:     "requester-job-selection-data-locality",
 		ConfigPath:   types.NodeRequesterJobSelectionPolicyLocality,
 		DefaultValue: Default.Node.Requester.JobSelectionPolicy.Locality,
 		Description:  `Only accept jobs that reference data we have locally ("local") or anywhere ("anywhere").`,
 	},
 	{
-		FlagName:     "job-selection-reject-stateless",
+		FlagName:     "requester-job-selection-reject-stateless",
 		ConfigPath:   types.NodeRequesterJobSelectionPolicyRejectStatelessJobs,
 		DefaultValue: Default.Node.Requester.JobSelectionPolicy.RejectStatelessJobs,
 		Description:  `Reject jobs that don't specify any data.`,
 	},
 	{
-		FlagName:     "job-selection-accept-networked",
+		FlagName:     "requester-job-selection-accept-networked",
 		ConfigPath:   types.NodeRequesterJobSelectionPolicyAcceptNetworkedJobs,
 		DefaultValue: Default.Node.Requester.JobSelectionPolicy.AcceptNetworkedJobs,
 		Description:  `Accept jobs that require network access.`,
 	},
 	{
-		FlagName:     "job-selection-probe-http",
+		FlagName:     "requester-job-selection-probe-http",
 		ConfigPath:   types.NodeRequesterJobSelectionPolicyProbeHTTP,
 		DefaultValue: Default.Node.Requester.JobSelectionPolicy.ProbeHTTP,
 		Description:  `Use the result of a HTTP POST to decide if we should take on the job.`,
 	},
 	{
-		FlagName:     "job-selection-probe-exec",
+		FlagName:     "requester-job-selection-probe-exec",
 		ConfigPath:   types.NodeRequesterJobSelectionPolicyProbeExec,
 		DefaultValue: Default.Node.Requester.JobSelectionPolicy.ProbeExec,
 		Description:  `Use the result of a exec an external program to decide if we should take on the job.`,
 	},
+	//
+	// Compute Job Selection Policy
+	{
+		FlagName:     "compute-job-selection-data-locality",
+		ConfigPath:   types.NodeComputeJobSelectionLocality,
+		DefaultValue: Default.Node.Compute.JobSelection.Locality,
+		Description:  `Only accept jobs that reference data we have locally ("local") or anywhere ("anywhere").`,
+	},
```

Comment:
this breaks backward compatibility and will need to update our documentation with the new flags. Also will be confusing for users when running hybrid nodes where defining `compute-job-selection-accept-networked` won't be enough as they will have to set `requester-job-selection-accept-networked` as well

The requirement here is to have a single flag that enables the job selection policy regardless if the node is a requester, compute or hybrid. I believe the quick fix for now as I understood during yesterday's meeting is to have the flag set compute node config instead of the requester, until we come up with a better solution of a single flag that sets multiple configs

Reply:
Agreed, we should be the ones doing the hard work to use the flag in two places, not the user.

Reply:
I'm guessing the worry about backwards compatibility isn't a deal-breaker here, since this isn't about changing the protocol, just the user experience. If compatibility is a showstopper, we might as well close this now. But if not, let's continue below:

Agreed that if we accept this change I'll need to update the documentation. I am happy to do that if this is approved.

> I believe the quick fix for now as I understood during yesterday's meeting is to have the flag set compute node config instead of the requester

I believe that is what's happening here, there is a single flag (set of flags) to configure the compute nodes policy. There is a different set of flags to configure the requesters policy, but that latter flag is ineffectual due to https://github.com/bacalhau-project/bacalhau/issues/2929

Here's why I chose to use two separate flags:
- It felt messy to use one flag to set up either my compute node, requester node, or both based on the type of node I had. 
  - I'm not 100% sure, but there might come a time when we want a hybrid node with different config values for its job selection policies: what if the requester and compute nodes need different settings for `probe-exec` or `probe-http` to decide if they'll take a job? A single flag won’t cut it then.
  - Also, the `data-locality` feels more like it’s meant for the compute node. Maybe we'll phase it out from the requester node at some point?
- At the moment, both compute and requester nodes have their unique config values. Using viper+cobra, there's no simple way to tie one flag to two different configs. Usually, one flag matches one config value.

But if we're set on using just one flag, we might consider creating a shared config that both node types can use. This would have common settings and flags for both nodes. I think there is a reasonable case for making these separate config values, but I'll implement whatever decision you decide on.